### PR TITLE
Add cert delete API and CLI command

### LINKED ISSUE #2849: Missing kontena certificate delete
There is no `kontena certificate delete` command to have the server forget about a certificate that is no longer in use by any services.

This is actually important, because the server certificate autorenewal could end up chewing through the LetsEncrypt rate limits with a lot of old, unused certificates.

### PR DESCRIPTION
**Pull request by jnummelin**
fixes #2849

### CODE REVIEW

File: server/app/routes/v1/certificates_api.rb [hunk 1]
```diff
@@ -73,6 +73,20 @@ def register(data)
           end
         end
 
+        r.delete do
+          r.on ':subject' do |subject|
+            @certificate = @grid.certificates.find_by(subject: subject)
+            if @certificate
+              @certificate.destroy
```
Comment (SpComb) [inline server/app/routes/v1/certificates_api.rb]: This needs to be a mutation that checks that the certificate is not in use by any services.

Comment (jnummelin) [reply server/app/routes/v1/certificates_api.rb]: true.

Comment (jnummelin) [reply server/app/routes/v1/certificates_api.rb]: fixed

File: server/app/mutations/grid_certificates/remove_certificate.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    include Common
+    include Logging
+
+    required do
+      model :certificate, class: Certificate
+    end
+
+    def validate
+      services_using = certificate.grid.grid_services.where(:'certificates.subject' => certificate.subject).map { |s| s.to_path}
+      add_error(self.certificate.subject, :certificate_in_use, "Certificate still in use in services: #{services_using.join(',')}") if services_using.size > 0
```
Comment (SpComb) [inline server/app/mutations/grid_certificates/remove_certificate.rb]: Bad idea to use the actual subject as the error key, it should be something constant like `:certificate` instead.

```
 [fail] Removing certificate for test.188.226.131.23.xip.io from development grid      
 [error] 422 : Unprocessable Entity:
	test:
	  '188':
	    '226':
	      '131':
	        '23':
	          xip:
	            io: 'Certificate still in use in services: development/ingress-lb/lb'
```

Comment (jnummelin) [reply server/app/mutations/grid_certificates/remove_certificate.rb]: oh, totally forgot the dotted functionality there.

Comment (SpComb) [reply server/app/mutations/grid_certificates/remove_certificate.rb]: Replace `add_error(self.certificate.subject, ...)` with just `add_error(:certificate, ...)`

File: server/spec/api/v1/certificates_spec.rb [hunk 3]
```diff
@@ -86,5 +84,30 @@
...
+        expect(response.status).to eq(200)
+      }.to change{Certificate.count}.by (-1)
+
+    end
+
+    it 'fails deleting certificate as it\'s in use' do
+      GridService.create!(grid: grid, name: 'redis', image_name: 'redis', certificates: [GridServiceCertificate.new(subject: 'kontena.io', name: 'SSL_CERT')])
+      expect {
+        delete "/v1/certificates/#{grid.name}/kontena.io", nil, request_headers
+        expect(response.status).to eq(422)
```
Comment (SpComb) [inline server/spec/api/v1/certificates_spec.rb]: Wouldn't hurt to also `expect(json_response['error']).to eq ....` here

**Comment by jnummelin** [comment]
fixed the error key

**Pull request merged**